src/m365/spp/commands/model/model-list.ts

```diff
@@ -0,0 +1,77 @@
+import { Logger } from '../../../../cli/Logger.js';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import { odata } from '../../../../utils/odata.js';
+import { spp, SppModel } from '../../../../utils/spp.js';
+import { urlUtil } from '../../../../utils/urlUtil.js';
+import { validation } from '../../../../utils/validation.js';
+import SpoCommand from '../../../base/SpoCommand.js';
+import commands from '../../commands.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+}
+
+class SppModelListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.MODEL_LIST;
+  }
+
+  public get description(): string {
+    return 'Retrieves the list of unstructured document processing models';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['AIBuilderHybridModelType', 'ContentTypeName', 'LastTrained', 'UniqueId', 'PublicationType'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initOptions();
+    this.#initTypes();
+    this.#initValidators();
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --siteUrl <siteUrl>'
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('siteUrl');
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        return validation.isValidSharePointUrl(args.options.siteUrl);
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (this.verbose) {
+        await logger.log(`Retrieving models from ${args.options.siteUrl}...`);
+      }
+
+      const siteUrl = urlUtil.removeTrailingSlashes(args.options.siteUrl);
+      await spp.assertSiteIsContentCenter(siteUrl);
+
+      const result = await odata.getAllItems<SppModel>(`${siteUrl}/_api/machinelearning/models`);
```

Comment:
Since we are not really doing anything with the result, I suggest that we type it as `any`. We can always add the model when we need it in another command.

Reply:
I will use uniqueId in `spp model get` command. But as I understand I should add only the used property there.